### PR TITLE
Modify vmware_vmx and docker acceptance tests to use builderT framework

### PR DESCRIPTION
**Pull request by SwampDragons**
This fixes the vmware acceptance tests to use the builderT framework instead of a home-rolled reimplementation of that framework. 

Does the same thing for the acceptance tests for the docker communicator. 

It also adds the file and shell provisioners to the builderT framework, to allow more complex builder tests. 

Finally, it moves the no-op progress tracker into the sdk, and uses it when defining the test ui in the builderT package. Adding the noop tracker to the test ui in the acceptance test framework was necessary to fix docker tests, which have been broken on main for some time.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -144,7 +144,7 @@ provisioners-acctest: install-build-deps generate
 	ACC_TEST_BUILDERS=$(ACC_TEST_BUILDERS) ACC_TEST_PROVISIONERS=$(ACC_TEST_PROVISIONERS) go test ./provisioner/... -timeout=1h
 
 # testacc runs acceptance tests
-testacc: install-build-deps generate ## Run acceptance tests
+testacc: # install-build-deps generate ## Run acceptance tests
```
Comment (nywilken) [inline Makefile:147]: I think removing this in testacc is good :+1:

File: builder/alicloud/ecs/builder_acc_test.go [hunk 2]
```diff
@@ -8,7 +8,7 @@ import (
 	"testing"
 
 	"github.com/aliyun/alibaba-cloud-sdk-go/services/ecs"
-	builderT "github.com/hashicorp/packer/packer-plugin-sdk/builder/testing"
+	builderT "github.com/hashicorp/packer/packer-plugin-sdk/acctest"
```
Comment (nywilken) [inline builder/alicloud/ecs/builder_acc_test.go:11]: This reads very nicely.

File: packer-plugin-sdk/acctest/testing.go [hunk 3]
```diff
@@ -123,6 +125,10 @@ func Test(t TestT, c TestCase) {
 					return nil, nil
 				},
 			},
+			ProvisionerStore: packersdk.MapOfProvisioner{
```
Comment (nywilken) [inline packer-plugin-sdk/acctest/testing.go:128]: :100: :100:

**Review by nywilken — APPROVED** [review]
LGTM

File: Makefile (hunk 1 above)
Comment (SwampDragons) [reply Makefile:147]: Not sure if we should remove it permanently; I'm still working on this branch (sorry, shoulda set it as a draft) and got sick of it regenerating every time.

File: builder/alicloud/ecs/builder_acc_test.go (hunk 2 above)
Comment (SwampDragons) [reply builder/alicloud/ecs/builder_acc_test.go:11]: yeah I think it's a lot clearer.

File: packer-plugin-sdk/acctest/provisioneracc/provisioners.go [hunk 4]
```diff
@@ -9,11 +9,10 @@ import (
 	"strings"
 	"testing"
 
-	testshelper "github.com/hashicorp/packer/helper/tests"
-
 	amazonEBS "github.com/hashicorp/packer/builder/amazon/ebs/acceptance"
 	virtualboxISO "github.com/hashicorp/packer/builder/virtualbox/iso/acceptance"
 	"github.com/hashicorp/packer/command"
+	"github.com/hashicorp/packer/packer-plugin-sdk/acctest/testutils"
```
Comment (nywilken) [inline packer-plugin-sdk/acctest/provisioneracc/provisioners.go:15]: Nice :100:

**Review by nywilken — APPROVED** [review]
LGTM :+1: :+1:

**Pull request merged**